lib/Db/PimcoreExtensionsTrait.php

```diff
@@ -611,4 +611,25 @@ public function setAutoQuoteIdentifiers($autoQuoteIdentifiers)
     {
         $this->autoQuoteIdentifiers = $autoQuoteIdentifiers;
     }
+
+    /**
+     * @param string $table
+     * @param string $idColumn
+     * @param string $where
+     */
+    public function selectAndDeleteWhere($table, $idColumn = 'id', $where = '')
+    {
+        $sql = 'SELECT ' . $idColumn . '  FROM ' . $table;
+
+        if ($where) {
+            $sql .= ' WHERE ' . $where;
+        }
+
+        $idsForDeletion = $this->fetchCol($sql);
+
+        if (!empty($idsForDeletion)) {
+            $idString = implode(",", $idsForDeletion);
```

Comment:
We should quote the IDs in case they are alphanumeric. 
`$idString = implode(',', array_map([$this, 'quote'], $idsForDeletion))`